File: src/algorithms/hash-maps/prefix-sum/subarray-sum-equals-k/SubarraySumEqualsK_test.java
public class SubarraySumEqualsK_test {
    public static void main(String[] args) {
        assert SubarraySumEqualsK.subarraySumEqualsK(new int[]{1, 1, 1}, 2) == 2;
        assert SubarraySumEqualsK.subarraySumEqualsK(new int[]{1, 2, 3}, 3) == 2;
        assert SubarraySumEqualsK.subarraySumEqualsK(new int[]{1, 2, 3}, 10) == 0;
        assert SubarraySumEqualsK.subarraySumEqualsK(new int[]{5}, 5) == 1;
        assert SubarraySumEqualsK.subarraySumEqualsK(new int[]{5}, 3) == 0;
        assert SubarraySumEqualsK.subarraySumEqualsK(new int[]{1, -1, 1}, 1) == 3;
        assert SubarraySumEqualsK.subarraySumEqualsK(new int[]{1, 2, 3, 4}, 10) == 1;
        assert SubarraySumEqualsK.subarraySumEqualsK(new int[]{0, 0, 0}, 0) == 6;
        assert SubarraySumEqualsK.subarraySumEqualsK(new int[]{2, 2, 2, 2}, 4) == 3;
        assert SubarraySumEqualsK.subarraySumEqualsK(new int[]{1, -1, 2, -2}, 0) == 3;

        System.out.println("All tests passed!");
    }
}

File: src/algorithms/arrays/prefix-sum/subarray-sum-equals-k/SubarraySumEqualsK_test.cpp
#include "sources/SubarraySumEqualsK.cpp"
#include <cassert>
#include <vector>
#include <iostream>

int main() {
    assert(subarraySumEqualsK({1, 2, 3}, 3).first == 2);
    assert(subarraySumEqualsK({1, 2, 3}, 10).first == 0);
    assert(subarraySumEqualsK({5, 1, 3}, 5).first == 1);
    assert(subarraySumEqualsK({}, 3).first == 0);
    assert(subarraySumEqualsK({3, 3, 3}, 3).first == 3);
    assert(subarraySumEqualsK({0, 0, 0}, 0).first == 6);
    assert(subarraySumEqualsK({7}, 7).first == 1);
    assert(subarraySumEqualsK({4}, 7).first == 0);

    std::cout << "All tests passed!" << std::endl;
    return 0;
}
